src/Raven.Server/Documents/Queries/AST/Query.cs

```diff
@@ -812,7 +812,9 @@ public IEnumerable<object> GetFinalValues()
                     throw new ArgumentOutOfRangeException("Unknown aggregation operation: " + Aggregation);
             }
 
-            return _values.Cast<object>();
+            return scale.HasValue 
+                ? _values.Select(v => (object)(v * scale.Value)) 
```

Comment:
Can we avoid LINQ here and use local static method instead (for both of the cases) ?

Reply:
Or can we scale it while we are adding the results?

Reply:
Scaling while adding is how I would like to go about it.

Reply:
done